framework/src/base/BoundaryRestrictable.C

```diff
@@ -256,10 +256,10 @@ BoundaryRestrictable::hasBoundaryMaterialPropertyHelper(const std::string & prop
     std::set<std::string> declared_props;
 
     // If boundary materials exist, populated the set of properties that were declared
-    if (material_warehouse.hasBoundaryMaterials(*id_it))
+    if (warehouse.hasActiveBoundaryObjects(*id_it))
     {
-      std::vector<Material *> mats = material_warehouse.getBoundaryMaterials(*id_it);
-      for (std::vector<Material *>::iterator mat_it = mats.begin(); mat_it != mats.end(); ++mat_it)
+      std::vector<MooseSharedPointer<Material> > mats = warehouse.getActiveBoundaryObjects(*id_it);
```

Comment:
Seems like you are pretty good at switching these to `const`. You either missed one, or I'm failing to see why this one can't be `const`